cmd/add/add.go

```diff
@@ -8,76 +8,86 @@ package add
 
 import (
 	"bufio"
+	"errors"
 	"fmt"
-	l "log"
 	"math"
 	"os"
 	"path"
+	"strings"
 
+	"github.com/NicksPatties/sweet/util"
 	"github.com/spf13/cobra"
 )
 
 var Cmd = &cobra.Command{
 	Use:   "add [flags] path",
 	Short: "Add an exercise",
-	Run: func(cmd *cobra.Command, args []string) {
-		if len(args) == 0 {
-			l.Fatalf("No path specified. Exiting.")
-		} else if len(args) > 1 {
-			l.Printf("Multiple paths specified. Ignoring paths %s", args[1:])
-		}
-		path := args[0]
-		start, _ := cmd.Flags().GetInt("start")
-		end, _ := cmd.Flags().GetInt("end")
-
-		// ... the rest of the command
-		addExercise(path, start, end)
+	Args:  cobra.ExactArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return addExercise(cmd, args)
 	},
 }
 
-func init() {
-	Cmd.Flags().IntP("start", "s", 1, "The start line number to extract the sample")
-	Cmd.Flags().IntP("end", "e", math.MaxUint32, "The end line number to extract the sample")
-}
+func addExercise(cmd *cobra.Command, args []string) (err error) {
+	if len(args) != 1 {
+		return errors.New("Incorrect number of args")
+	}
+	pathName := args[0]
+	start, _ := cmd.Flags().GetUint("start")
+	end, _ := cmd.Flags().GetUint("end")
+
+	if start > end {
+		fmt.Printf("YOU ARE BEING BAD!!!!")
+		return errors.New("start flag cannot be greater than end flag")
+	}
 
-// Adds an exercise to sweet's configured exercises directory.
-func addExercise(pathName string, start int, end int) {
-	// Open the exercise from the given path.
 	inputFile, err := os.Open(pathName)
 	if err != nil {
-		fmt.Println(err)
-		fmt.Println("Exiting.")
-		os.Exit(1)
+		return
 	}
 	defer inputFile.Close()
 
 	// Create the new exercise file in the configuration directory.
-	newExercisePath := path.Join("/home/nick/.config/sweet/exercises", path.Base(pathName))
+	sweetConfigDir, err := util.SweetConfigDir()
+	if err != nil {
+		return
+	}
+	exercisesDir := path.Join(sweetConfigDir, "exercises")
+	if envDir := os.Getenv("SWEET_EXERCISES_DIR"); envDir != "" {
+		exercisesDir = envDir
+	}
+	newExercisePath := path.Join(exercisesDir, path.Base(pathName))
 	newExerciseFile, err := os.Create(newExercisePath)
```

Comment:
Suppose I want to do this
```sh
sweet add file.go -s 3 -e 6
sweet add file.go -s 10 -e 20
```
What files will get saved? Will the file I want to write be overwritten? I should be able to save different exercises from the same file with no issues.

Reply:
What should the name of the files be when I provide:
- if start is not equal to 1 and end is not equal to length of exercise: `file#{start}-{end}.{extension}` 
  - extension being at the end helps determine which language is being used
- if start is at 1 and end is equal to len(exercise), don't change the filename

Reply:
This sounds like a feature request, but for now, I'll error if the file already exists when I add it.